puree/build.gradle

```diff
@@ -140,46 +88,66 @@ publishing {
 
                 def license = root.appendNode('licenses').appendNode('license')
                 license.appendNode('name', 'MIT')
-                license.appendNode('url',
-                        'https://raw.githubusercontent.com/mercari/puree-android/master/LICENSE.txt')
+                license.appendNode('url', Publishing.licenseUrl)
                 license.appendNode('distribution', 'repo')
+
+                // Add all dependencies in pom metadata
+                final dependenciesNode = root.appendNode('dependencies')
+
+                ext.addDependency = { Dependency dep, String scope ->
+                    if (dep.group == null || dep.version == null || dep.name == null || dep.name == "unspecified")
+                        return // ignore invalid dependencies
+
+                    final dependencyNode = dependenciesNode.appendNode('dependency')
+                    dependencyNode.appendNode('groupId', dep.group)
+                    dependencyNode.appendNode('artifactId', dep.name)
+                    dependencyNode.appendNode('version', dep.version)
+                    dependencyNode.appendNode('scope', scope)
+
+                    if (!dep.transitive) {
+                        // If this dependency is transitive, we should force exclude all its dependencies them from the POM
```

Comment:
Is this backwards? The code says "if not transitive", but the comment says "if transitive"

Reply:
That's an old setup I'm using in Android projects, so not sure about that. But after checking a similar project implementation:
https://github.com/StefMa/AndroidArtifacts/blob/72e9de7ab909a31764ec5e01a9be00f59e48181b/src/main/kotlin/guru/stefma/androidartifacts/MavenPom.kt

Add specified exclude rules make sense. On the other hand, not sure this one is helpful.
Let me investigate what could happen if we're dropping this section.

Reply:
I checked before/after generated pom and output didn't change as expected. I remove these useless lines 👍